app/code/Magento/Inventory/Model/ReservationBuilder/Validator/StockIdValidator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model\ReservationBuilder\Validator;
+
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\Inventory\Model\ReservationBuilderInterface;
+
+/**
+ * Check that stock ID is valid
+ */
+class StockIdValidator implements ReservationBuilderValidatorInterface
+{
+    /**
+     * @var ValidationResultFactory
+     */
+    private $validationResultFactory;
+
+    /**
+     * @param ValidationResultFactory $validationResultFactory
+     */
+    public function __construct(ValidationResultFactory $validationResultFactory)
+    {
+        $this->validationResultFactory = $validationResultFactory;
+    }
+
+    public function validate(ReservationBuilderInterface $reservationBuilder): ValidationResult
+    {
+        $errors = [];
+
+        $value = $reservationBuilder->getStockId();
+
+        if (false === is_numeric($value)) { // TODO should also check if it is positive?
```

Comment:
potentially we could check whether provided StockId exists in the system.
But that will introduce performance degradation. Because we need to make additional service call for each builder::build

So, let's leave as-is now